src/coreclr/vm/jitinterface.cpp

```diff
@@ -715,6 +715,52 @@ int CEEInfo::getStringLiteral (
     return result;
 }
 
+int CEEInfo::objectToString (
+        void*     handle,
+        char16_t* buffer,
+        int       bufferSize)
+{
+    CONTRACTL{
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    int result = -1;
+
+    _ASSERT(handle != nullptr && buffer != nullptr && bufferSize > 0);
+
+    JIT_TO_EE_TRANSITION();
+
+    Object* obj = (Object*)handle;
+
+    GCX_COOP();
+
+    StackSString stackStr;
+
+    // Currently only supported for String and RuntimeType
+    if (obj->GetMethodTable()->IsString())
+    {
+        ((StringObject*)obj)->GetSString(stackStr);
+    }
+    else if (obj->GetMethodTable() == g_pRuntimeTypeClass)
+    {
+        ((ReflectClassBaseObject*)obj)->GetType().GetName(stackStr);
+    }
+
+    if (stackStr.GetCount() > 0)
+    {
+        result = min(bufferSize, (int)stackStr.GetCount());
+        memcpy((BYTE*)buffer, stackStr.GetUnicode(), result * 2);
+        // Null-terminate it (trim if needed)
+        buffer[result == bufferSize ? result - 1 : result] = 0;
+    }
```

Comment:
This is still missing null termination for the empty string case.

Reply:
@jakobbotsch the buffer is zeroed on jit side so technically it's not but I assume it's better to handle it here

Reply:
I think so too. Otherwise it makes the contract pretty strange/inefficient.

Reply:
Fixed, thanks!

Reply:
checked that it works:
```asm
       48B80800AD1144020000 mov      rax, 0x24411AD0008      ; ''
```